test/units/testsuite-23.oneshot-restart.sh

```diff
@@ -48,5 +51,32 @@ sleep 5
 if [[ $(cat $TMP_FILE) != "aaa" ]]; then
     exit 1
 fi
+rm "$TMP_FILE"
+
+# Test RestartForceExitStatus=. Note that success exit statuses are meant to be skipped
+
+TMP_FILE="/tmp/test-23-oneshot-restart-test$RANDOM"
+UNIT_NAME="testsuite-23-oneshot-restartforce.service"
+
+cat >"/run/systemd/system/$UNIT_NAME" <<EOF
+[Service]
+Type=oneshot
+RestartForceExitStatus=0 2
+ExecStart=/usr/lib/systemd/tests/testdata/testsuite-23.units/testsuite-23-oneshot-restartforce.sh "$TMP_FILE"
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+# Pin the unit in memory
+systemctl enable "$UNIT_NAME"
+(! systemctl start "$UNIT_NAME")
+sleep 5
```

Comment:
let's not add more sleep based tests, they tend to timeout under load.

my proposal: use a fifo in the fs. they have the nice property to block when you open them for read if noone is connected on the other side. hence for the first run, echo something into the fifo and wait for it via shell "read". and then maybe use OnSuccess= or so to fire another unit that reports via another fifo that the first one finished. by waiting for that you know that the unit is properly complete.

Reply:
> hence for the first run, echo something into the fifo and wait for it via shell "read"

Hmm, I don't see why this one is needed. Updated with a slightly different approach.